cpp/src/join/conditional_join.cu

```diff
@@ -118,8 +134,14 @@ conditional_join(table_view const& left,
 
   // If the output size will be zero, we can return immediately.
   if (join_size == 0) {
-    return std::make_pair(std::make_unique<rmm::device_uvector<size_type>>(0, stream, mr),
-                          std::make_unique<rmm::device_uvector<size_type>>(0, stream, mr));
+    auto join_indices{
+      std::make_pair(std::make_unique<rmm::device_uvector<size_type>>(0, stream, mr),
+                     std::make_unique<rmm::device_uvector<size_type>>(0, stream, mr))};
+    if (join_type == join_kind::FULL_JOIN) {
+      auto complement_indices = detail::get_left_join_indices_complement(
+        join_indices.second, left.num_rows(), right.num_rows(), stream, mr);
+      return detail::concatenate_vector_pairs(join_indices, complement_indices, stream);
+    }
```

Comment:
Here I only see "If the output size is zero AND join_type == join_kind::FULL_JOIN, we can return immediately" .

Reply:
Wow thank you for bothering me more on this, clearly my last fix was hasty. I've added a much more descriptive comment of what the behavior _should_ be and updated the code to match.